nibabel/bv.py

```diff
@@ -0,0 +1,691 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+# ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+# ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Reading / writing functions for Brainvoyager (BV) file formats.
+
+please look at the support site of BrainInnovation for further informations
+about the file formats: http://support.brainvoyager.com/
+
+This file implements basic functionality for BV file formats. Look into bv_*.py
+files for implementations of the different file formats.
+
+Author: Thomas Emmerling
+"""
+
+import numpy as np
+from .volumeutils import array_to_file, array_from_file
+from .spatialimages import Header, HeaderDataError, SpatialImage
+from .fileholders import copy_file_map
+from .arrayproxy import CArrayProxy
+from .volumeutils import make_dt_codes
+from struct import pack, unpack, calcsize
+
+_dtdefs = (  # code, conversion function, equivalent dtype, aliases
+    (1, 'int16', np.uint16),
+    (2, 'float32', np.float32),
+    (3, 'uint8', np.uint8))
+
+# Make full code alias bank, including dtype column
+data_type_codes = make_dt_codes(_dtdefs)
+
+
+def readCString(f, nStrings=1, bufsize=1000, startPos=None, strip=True,
+                rewind=False):
+    """Read a zero-terminated string from a file object.
+
+    Read and return a zero-terminated string from a file object.
+
+    Parameters
+    ----------
+    f : fileobj
+       File object to use
+    nStrings: int, optional
+       Number of strings to search (and return). Default is 1.
+    bufsize: int, optional
+       Define the buffer size that should be searched for the string.
+       Default is 1000 bytes.
+    startPos: int, optional
+       Define the start file position from which to search. If None then start
+       where the file object currently points to. Default is None.
+    strip : bool, optional
+       Whether to strip the trailing zero from the returned string.
+       Default is True.
+    rewind: bool, optional
+       Whether the fileobj f should be returned to the initial position after
+       reading. Default is False.
+    Returns
+    -------
+    [str] : list of string(s)
+    """
+    currentPos = f.tell()
+    if strip:
+        suffix = ''
+    else:
+        suffix = '\x00'
+    if startPos is not None:
+        f.seek(startPos)
+    data = f.read(bufsize)
+    lines = data.split('\x00')
+    if rewind:
+        f.seek(currentPos)
+    else:
+        offset = 0
+        for s in range(nStrings):
+            offset += len(lines[s])+1
+        f.seek(currentPos+offset)
+    for s in range(nStrings):
+        yield lines[s] + suffix
```

Comment:
You said in the returns section this returns a list, but in fact this is a generator.   I guess no real need for generator here?  But if you do want a generator, you could rename the function and rewrite docstring.